google/cloud/testing_util/chrono_matchers.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_CHRONO_MATCHERS_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_TESTING_UTIL_CHRONO_MATCHERS_H
+
+#include "google/cloud/internal/type_traits.h"
+#include "google/cloud/version.h"
+#include "absl/time/time.h"
+#include <chrono>
+#include <ostream>
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+
+template <class Rep, class Period>
+constexpr bool kIsChronoDurationOstreamable =
+    google::cloud::internal::IsOStreamable<
+        std::chrono::duration<Rep, Period>>::kValue;
+
+template <class Clock, class Duration>
+constexpr bool kIsChronoTimePointOstreamable =
+    google::cloud::internal::IsOStreamable<
+        std::chrono::time_point<Clock, Duration>>::kValue;
+
+}  // namespace internal
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_END
+}  // namespace cloud
+}  // namespace google
+
+/**
+ * Provide `std::chrono::duration` and `std::chrono::time_point` output
+ * streaming when otherwise unsupported. This enables the printing of
+ * legible chrono values from googletest, rather than raw bytes, when a
+ * match fails.
+ *
+ * It is strictly undefined behavior to add to any `std` namespace, but
+ * it works, and, in this test-only case, is well worth it.
+ */
+namespace std {
```

Comment:
If you are willing to open the namespace, then adding `PrintTo()` overloads is also an option:

https://github.com/google/googletest/blob/main/docs/advanced.md#teaching-googletest-how-to-print-your-values

Unless you worry about a `PrintTo()` already in the `std::chrono()` namespace

Reply:
The win with the existing incantation is that normal, outside-googletest streaming also works.  I admit that that whole thing is currently wrapped in "chrono_matcher" clothing, but having it also for non-googletest tests is nice.  We could change the filename.

Reply:
Ack.  Your call I think.

Reply:
Renamed to "chrono_output".